src/compiler/checker.ts

```diff
@@ -4165,7 +4165,19 @@ namespace ts {
                 return getTypeForBindingElement(<BindingElement>declaration);
             }
 
-            const isOptional = !isBindingElement(declaration) && !isVariableDeclaration(declaration) && !!declaration.questionToken && includeOptionality;
+            let isOptional = false;
+            if (includeOptionality) {
+                if (isInJavaScriptFile(declaration) && isParameterDeclaration(declaration)) {
+                    const parameterTags = getJSDocParameterTags(declaration as ParameterDeclaration);
+                    if (parameterTags && parameterTags.length > 0 && find(parameterTags, tag => tag.isBracketed)) {

```

Comment:
Simpler to just write `isOptional = ...` than `if (...) isOptional = true;

Reply:
I guess so. Doesn't match the next branch, so it makes the code a little less obvious to read.